lib/game.rb

```diff
@@ -0,0 +1,187 @@
+
+require_relative 'board'
+
+class Game
+  attr_reader :board, :active_player
+
+  def initialize
+    @board = Board.new
+    @active_player = :white
+    @players = %i[white black]
+  end
+
+  def compute_path(figure, start, stop)
+    path = [start]
+    if figure.is_a?(Knight)
+      path.push(stop)
+      return path
+    end
+    current = start
+    rows = stop[0] - start[0]
+    columns = stop[1] - start[1]
+    if rows.zero? && columns.positive?
+      until current == stop
+        current = [current[0], (current[1] + 1)]
+        path.push(current)
+      end
+    elsif rows.zero? && columns.negative?
+      until current == stop
+        current = [current[0], (current[1] - 1)]
+        path.push(current)
+      end
+    elsif rows.positive? && columns.zero?
+      until current == stop
+        current = [(current[0] + 1), current[1]]
+        path.push(current)
+      end
+    elsif rows.negative? && columns.zero?
+      until current == stop
+        current = [(current[0] - 1), current[1]]
+        path.push(current)
+      end
+    elsif rows.positive? && columns.positive?
+      until current == stop
+        current = [(current[0] + 1), (current[1] + 1)]
+        path.push(current)
+      end
+    elsif rows.positive? && columns.negative?
+      until current == stop
+        current = [(current[0] + 1), (current[1] - 1)]
+        path.push(current)
+      end
+    elsif rows.negative? && columns.positive?
+      until current == stop
+        current = [(current[0] - 1), (current[1] + 1)]
+        path.push(current)
+      end
+    elsif rows.negative? && columns.negative?
+      until current == stop
+        current = [(current[0] + - 1), (current[1] - 1)]
+        path.push(current)
+      end
+    end
+    path
+  end
+
+  def enemy?(row, column)
+    color = if @active_player == :white
+              :black
+            else
+              :white
+            end
+    figure = board.find_figure(color, row, column)
+    !figure.nil?
+  end
+
+  def no_obstacles?(path)
+    # Get rid of starting spot
+    path.shift
+    # Get rid of ending point and return false if there is friendly figure on it
+    last_tile = path.pop
+    unless enemy?(last_tile[0], last_tile[1]) || @board.find_figure(@active_player, last_tile[0], last_tile[1]).nil?
+      return false
+    end
+
+    path.each do |tile|
+      unless @board.find_figure(:white, tile[0], tile[1]).nil? && @board.find_figure(:black, tile[0], tile[1]).nil?
+        return false
+      end
+    end
+    true
+  end
+
+  def valid_input?(input)
+    start = input[0..1]
+    stop = input[-2..]
+    if ('a'..'h').include?(start[0]) && ('a'..'h').include?(stop[0]) && ('1'..'8').include?(start[1]) && ('1'..'8').include?(stop[1])
+      return true
+    end
+
+    false
+  end
+
+  def move_possible?(figure, row, column)
+    possible_moves = figure.possible_moves
+    return false if possible_moves.nil?
+
+    return true if possible_moves.include?([row, column])
```

Comment:
Also, I will be adding points to your counter from now on, every time I see you writing a method that returns `true/false` explicitely in multiple steps.

```
   return possible_moves&.include([row, column])
```